edx_proctoring/api.py

```diff
@@ -899,10 +899,17 @@ def update_attempt_status(exam_id, user_id, to_status, raise_if_not_found=True,
             return_course_info=True
         )
 
-        send_proctoring_attempt_status_email(
-            exam_attempt_obj,
-            credit_state.get('course_name', _('your course'))
-        )
+        if credit_state:
+            send_proctoring_attempt_status_email(
+                exam_attempt_obj,
+                credit_state.get('course_name', _('your course'))
+            )
+        else:
+            log.error(
+                "Could not retrieve the credit state for the user with the user id %r in the course %r.",
```

Comment:
The credit service returns None if the student is not enrolled in the course, or the course doesn't have credit. I figured that it was a little difficult to fix this entirely without more information, which is why I decided that logging the issue would be more helpful for now.

I'll make the error message more specific now.